drivers/pinctrl/intel/pinctrl-intel.c

```diff
@@ -1292,6 +1324,17 @@ static int intel_pinctrl_pm_init(struct intel_pinctrl *pctrl)
 			return -ENOMEM;
 
 		communities[i].intmask = intmask;
+
+		if (dmi_first_match(dmi_retain_hostown_table)) {
```

Comment:
You don't need to call `dmi_first_match` for every iteration of this loop -- you can check for `PINCTRL_QUIRK_KEEP_HOSTOWN` here as well and set it in `intel_pinctrl_probe` before the call to `intel_pinctrl_pm_init` (sorry for not catching this on the previous round).